Sources/Identity/CustomerInfo.swift

```diff
@@ -164,263 +153,170 @@ import Foundation
-            self.entitlementsData = subscriberData["entitlements"] as? [String: Any] ?? [:]
-            self.nonSubscriptionTransactions = TransactionsFactory.nonSubscriptionTransactions(
-                withSubscriptionsData: nonSubscriptionsByProductId,
-                dateFormatter: dateFormatter)
-
-            let latestNonSubscriptionTransactionsByProductId = [String: [String: Any]](
-                uniqueKeysWithValues: nonSubscriptionsByProductId.map { productId, transactionsArray in
-                    (productId, transactionsArray.last ?? [:])
-                })
-
-            self.allTransactionsByProductId = latestNonSubscriptionTransactionsByProductId
-                .merging(subscriptionTransactionsByProductId, strategy: .keepOriginalValue)
-
-            self.allPurchases = latestNonSubscriptionTransactionsByProductId
-                .merging(subscriptionTransactionsByProductId) { (current, _) in current }
-        }
+    static let currentSchemaVersion = "3"
+
+}
+
+extension CustomerInfo: RawDataContainer {
+
+    // Docs inherited from protocol
+    // swiftlint:disable missing_docs
+    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+    public var underlyingData: some Encodable {
+        return self.data.response
     }
 
 }
 
-extension CustomerInfo: RawDataContainer {}
-
-enum CustomerInfoError: Int, DescribableError {
-
-    case missingJsonObject = 0
-    case cantInstantiateJsonObject
-    case requestDateFromJson
-    case requestDateFromString
-
-    var description: String {
-        switch self {
-        case .missingJsonObject:
-            return "Unable to read property \"subscriber\" from json."
-        case .cantInstantiateJsonObject:
-            return "json appears to be in an unexpected format."
-        case .requestDateFromJson:
-            return "Unable to read property \"request_date\" from json."
-        case .requestDateFromString:
-            return "Unable to parse a date from json propery \"request_date\"."
+/// `CustomerInfo`'s `Codable` implementation relies on `Data`
+extension CustomerInfo: Codable {
+
+    // swiftlint:disable:next missing_docs
+    public convenience init(from decoder: Decoder) throws {
+        do {
+            self.init(data: try Data(from: decoder))
+        } catch {
+            throw ErrorUtils.customerInfoError(error: error)
         }
     }
 
+    // swiftlint:disable:next missing_docs
+    public func encode(to encoder: Encoder) throws {
+        try self.data.encode(to: encoder)
+    }
+
 }
 
+extension CustomerInfo: HTTPResponseBody {}
+
+// MARK: - Private
+
 private extension CustomerInfo {
 
-    static func createSubscriberDataError(_ error: Error, subscriberDictionary: [String: Any]) -> Error {
-        Logger.error(Strings.customerInfo.cant_instantiate_from_json_object(jsonObject: subscriberDictionary))
+    /// The actual contents of a `CustomerInfo`: the response with the associated version.
+    struct Data {
```

Comment:
Good point. The whole type is `CustomerInfo.Data`, maybe I can use that whole type for clarity?

Reply:
I think that works, although the reason I suggested renaming was so that in the future we don't use it by calling it `Data` and have this conversation again

Reply:
That's fair. Maybe `CustomerInfo.Contents`?

Reply:
Changed!